--- efl-1.10.1/src/modules/evas/loaders/gif/evas_image_load_gif.c.orig	2014-06-03 12:38:57.000000000 +0200
+++ efl-1.10.1/src/modules/evas/loaders/gif/evas_image_load_gif.c	2014-06-14 16:57:31.264950243 +0200
@@ -528,7 +528,11 @@
    ret = EINA_TRUE;
 
 on_error: // jump here on any errors to clean up
+#if (GIFLIB_MAJOR > 5) || ((GIFLIB_MAJOR == 5) && (GIFLIB_MINOR >= 1))
+   if (gif) DGifCloseFile(gif, NULL);
+#else
    if (gif) DGifCloseFile(gif);
+#endif
    if (fi.map) eina_file_map_free(f, fi.map);
    return ret;
 }
@@ -606,7 +610,11 @@
    // start as we have nothnig to build on
    if ((index > 0) && (index < loader->imgnum) && (animated->animated))
      {
+#if (GIFLIB_MAJOR > 5) || ((GIFLIB_MAJOR == 5) && (GIFLIB_MINOR >= 1))
+        if (loader->gif) DGifCloseFile(loader->gif, NULL);
+#else
         if (loader->gif) DGifCloseFile(loader->gif);
+#endif
         if ((loader->fi.map) && (loader->f))
           eina_file_map_free(loader->f, loader->fi.map);
         loader->gif = NULL;
@@ -764,7 +772,11 @@
    loader->imgnum = imgnum;
    if ((animated->frame_count <= 1) || (rec == TERMINATE_RECORD_TYPE))
      {
+#if (GIFLIB_MAJOR > 5) || ((GIFLIB_MAJOR == 5) && (GIFLIB_MINOR >= 1))
+        if (loader->gif) DGifCloseFile(loader->gif, NULL);
+#else
         if (loader->gif) DGifCloseFile(loader->gif);
+#endif
         if ((loader->fi.map) && (loader->f))
           eina_file_map_free(loader->f, loader->fi.map);
         loader->gif = NULL;
@@ -848,7 +860,11 @@
 evas_image_load_file_close_gif2(void *loader_data)
 {
    Loader_Info *loader = loader_data;
+#if (GIFLIB_MAJOR > 5) || ((GIFLIB_MAJOR == 5) && (GIFLIB_MINOR >= 1))
+   if (loader->gif) DGifCloseFile(loader->gif, NULL);
+#else
    if (loader->gif) DGifCloseFile(loader->gif);
+#endif
    if ((loader->fi.map) && (loader->f))
      eina_file_map_free(loader->f, loader->fi.map);
    free(loader);
