--- efl-1.8.1/src/modules/evas/engines/wayland_egl/evas_engine.c.orig	2013-11-24 05:18:45.000000000 +0100
+++ efl-1.8.1/src/modules/evas/engines/wayland_egl/evas_engine.c	2013-12-07 12:52:11.209362982 +0100
@@ -128,8 +128,6 @@
 /* external variables */
 int _evas_engine_wl_egl_log_dom = -1;
 
-#ifdef GL_GLES
-
 # ifndef EGL_NATIVE_PIXMAP_KHR
 #  define EGL_NATIVE_PIXMAP_KHR 0x30b0
 # endif
@@ -146,8 +144,6 @@
 unsigned int (*glsym_eglUnmapImageSEC) (void *a, void *b, int c) = NULL;
 unsigned int (*glsym_eglSwapBuffersWithDamage) (EGLDisplay a, void *b, const EGLint *d, EGLint c) = NULL;
 
-#endif
-
 /* local functions */
 static Eina_Bool 
 evgl_init(Render_Engine *re)
--- efl-1.8.1/src/modules/evas/engines/wayland_egl/evas_engine.h.orig	2013-11-04 12:58:09.000000000 +0100
+++ efl-1.8.1/src/modules/evas/engines/wayland_egl/evas_engine.h	2013-12-07 15:04:14.622363802 +0100
@@ -6,8 +6,8 @@
 /* NB: This already includes wayland-client.h */
 # include <wayland-egl.h>
 
+# include <EGL/egl.h>
 # ifdef GL_GLES
-#  include <EGL/egl.h>
 /* NB: These are already included from gl_common */
 /* #  include <GLES2/gl2.h> */
 /* #  include <GLES2/gl2ext.h> */
@@ -65,12 +65,10 @@
         Eina_Bool drew : 1;
      } draw;
 
-#ifdef GL_GLES
    EGLContext egl_context[1];
    EGLSurface egl_surface[1];
    EGLConfig egl_config;
    EGLDisplay egl_disp;
-#endif
 
    Eina_Bool surf : 1;
 };
--- efl-1.8.1/src/modules/evas/engines/wayland_egl/evas_wl_main.c.orig	2013-11-04 12:58:09.000000000 +0100
+++ efl-1.8.1/src/modules/evas/engines/wayland_egl/evas_wl_main.c	2013-12-07 16:55:11.888751090 +0100
@@ -170,8 +170,10 @@
         return NULL;
      }
 
+#ifdef GL_GLES
    gw->gl_context->egldisp = gw->egl_disp;
    gw->gl_context->eglctxt = gw->egl_context[0];
+#endif
 
    eng_window_use(gw);
    evas_gl_common_context_resize(gw->gl_context, w, h, rot);
